Support loading Python agent through sw-python CLI (#156)
apache/skywalking-python: docs/CLI.md

# SkyWalking Python Agent Command-Line Interface(CLI)

> You would at least need to add the following lines to your applications in previous releases to get the agent attached and running.  

Reviewer: What about putting the condition at the beginning? In case the users just read a half of the sentence 🤣 and let's be clear about since which version we support this

```suggestion
In earlier releases than 0.7.0, you would at least need to add the following lines to your applications to get the agent attached and running.  
```

```python
from skywalking import agent
agent.start()
```

Now the SkyWalking Python agent implements a command-line interface that can be utilized to attach the agent to your
awesome applications during deployment **without changing any application code**, 
just like the [SkyWalking Java Agent](https://github.com/apache/skywalking).

## Usage

> Upon successful installation of the SkyWalking Python agent via pip, 

Reviewer: ```suggestion
Upon successful [installation of the SkyWalking Python agent via pip](../README.md#install),
```

a command-line script `sw-python` is installed in your environment(virtual env preferred).

> ### The run option

Reviewer: ```suggestion
### The `run` option
```

Therefore, any attempt to pass a command that uses a different Python interpreter/ environment will not bring up 
SkyWalking Python Agent even if another SkyWalking Python agent is installed there(no matter the version), 
and will force exit with an error message indicating the reasoning.

#### Disabling child processes from starting new agents

Sometimes you don't actually need the agent to monitor anything in a child process.

If you do not need the agent to get loaded for application child processes, you can turn off the behavior by setting an environment variable.

`SW_PYTHON_BOOTSTRAP_PROPAGATE` to `False`

Note the auto bootstrap depends on the environment inherited by child processes, 
thus prepending a new sitecustomize path to or removing the loader path from the `PYTHONPATH` could prevent the agent from loading in a child process. 

### Configuring the agent 

You would normally want to provide additional configurations other than the default ones.

#### Through environment variables

The currently supported method is to provide the environment variables listed 
in [EnvVars Doc](EnvVars.md) as instructed in the [README](../README.md).

> #### Through a sw-config.yaml

Reviewer: One thing we should note when implementing this is that we've been trying to minimize our dependencies as an agent, so we'd better choose a reasonable file format that Python built-in libs can parse, instead of bringing another library, to avoid version conflicts with user applications, `yaml` is a common case where users may already use and we should not require dependency upgrade or downgrade.
